src/main/js/components/CnesReportApp.js

```diff
@@ -121,8 +146,60 @@ export default class CnesReportApp extends React.PureComponent {
+                            <input id="enableMdHidden" type="hidden" value="false" name="enableMd" disabled={this.state.enableMd}/>
+                            <input type="checkbox"
+                                id="enableMd"
+                                name="enableMd"
+                                value="true"
+                                defaultChecked={this.state.enableMd}
+                                onChange={() => this.onChangeCheckbox('enableMd')}/>
+                            <label for="enableMd" id="enableMdLabel"><strong>Enable MD generation</strong></label>
+                        </div>
+                        <div>
+                            <input id="enableXlsxHidden" type="hidden" value="false" name="enableXlsx" disabled={this.state.enableXlsx}/>
+                            <input type="checkbox"
+                                id="enableXlsx"
+                                name="enableXlsx"
+                                value="true"
+                                defaultChecked={this.state.enableXlsx}
+                                onChange={() => this.onChangeCheckbox('enableXlsx')}/>
+                            <label for="enableXlsx" id="enableXlsxLabel"><strong>Enable XLSX generation</strong></label>
+                        </div>
+                        <div>
+                            <input id="enableCsvHidden" type="hidden" value="false" name="enableCsv" disabled={this.state.enableCsv}/>
+                            <input type="checkbox"
+                                id="enableCsv"
+                                name="enableCsv"
+                                value="true"
+                                defaultChecked={this.state.enableCsv}
+                                onChange={() => this.onChangeCheckbox('enableCsv')}/>
+                            <label for="enableCsv" id="enableCsvLabel"><strong>Enable CSV generation</strong></label>
+                        </div>
+                        <div>
+                            <input id="enableConfHidden" type="hidden" value="false" name="enableConf" disabled={this.state.enableConf}/>
+                            <input type="checkbox"
+                                id="enableConf"
+                                name="enableConf"
+                                value="true"
+                                defaultChecked={this.state.enableConf}
+                                onChange={() => this.onChangeCheckbox('enableConf')}/>
+                            <label for="enableConf" id="enableConfLabel"><strong>Enable quality configuration generation</strong></label>
+                        </div>
+                        <br />
+                        <input id="generation" name="generation" type="submit" value="Generate"
+                            disabled={!(this.state.enableDocx || this.state.enableMd || this.state.enableXlsx || this.state.enableCsv || this.state.enableConf)}/>

```

Comment:
Maybe we can extract the logic here in a dedicated function so that we can explain (with comments) why we do this.